src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -242,6 +257,7 @@ def set_opts(self, curl, url, params, headers,
         if verbose:
             curl.setopt(pycurl.VERBOSE, True)
             curl.setopt(pycurl.DEBUGFUNCTION, self.debug)
+        self.request_headers = headers
```

Comment:
Given that this attribute is not used anywhere but here in the `__init__` method, it looks like this line could be removed.

Reply:
the request headers can be required in different clients, e.g. DBSClient, and should be set accodringly. This is why it was set here. The are initially declared in ctor and set appropriate for every HTTP request.